Sources/GRPC/Interceptor/ServerInterceptors.swift

```diff
@@ -42,7 +42,7 @@ import NIOCore
 /// require any extra attention. However, if work is done on a `DispatchQueue` or _other_
 /// `EventLoop` then implementers should ensure that they use `context` from the correct
 /// `EventLoop`.
-open class ServerInterceptor<Request, Response> {
+open class ServerInterceptor<Request, Response>: @unchecked Sendable {
```

Comment:
Can we actually do this? Making an `open class` @unchecked Sendable requires that all subclasses are also `Sendable` which they might not be.

Reply:
I'm not sure what other options we have: any subclasses of this _must_ be thread safe (as documented), we just didn't enforce it previously...

Reply:
Discussed this offline: Any subclass must be `Sendable` and this will just add warnings to adopters so not breaking any API